framework/src/systems/SystemBase.C

```diff
@@ -1426,9 +1426,14 @@ SystemBase::applyScalingFactors(const std::vector<Real> & inverse_scaling_factor
   for (MooseIndex(_vars) thread = 0; thread < _vars.size(); ++thread)
   {
     auto & field_variables = _vars[thread].fieldVariables();
-    for (MooseIndex(field_variables) i = 0; i < field_variables.size(); ++i)
-      field_variables[i]->scalingFactor(1. / inverse_scaling_factors[i] *
-                                        field_variables[i]->scalingFactor());
+    for (MooseIndex(field_variables) i = 0, p = 0; i < field_variables.size(); ++i)
+    {
+      auto factors = field_variables[i]->arrayScalingFactor();
```

Comment:
```suggestion
      const auto & factors = field_variables[i]->arrayScalingFactor();
```

Reply:
Intentionally making a copy here because we want to modify the factors with the auto generated factors and apply the modified factor.

Reply:
sounds good